apis/v1/team_types.go

```diff
@@ -0,0 +1,42 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Team is the collection of members of a team.
+type Team struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TeamSpec   `json:"spec,omitempty"`
+	Status TeamStatus `json:"status,omitempty"`
+}
+
+// TeamSpec contains the desired members of a team.
+type TeamSpec struct {
+	DisplayName string    `json:"displayName,omitempty"`
+	UserRefs    []UserRef `json:"userRefs"`
+}
+
+// TeamStatus contains the actual members of a team.
+type TeamStatus struct {
```

Comment:
That's not in the design doc, but a group id field in the status might help you when implementing the adapter. I thought about doing the same for the Organization (but it's a little more bastelig there, so I did not get to it)

Reply:
Haven't looked too deep into the mapping yet. Thanks for the tip 👍